UsingAuthentication.md

```diff
@@ -0,0 +1,108 @@
+
+```
+public class AuthFilter implements Filter {
+  @SuppressWarnings("unused")
+  private final String callingServiceName;
+
+  public AuthFilter(String callingServiceName) {
+    Validate.notNull(callingServiceName);
+    this.callingServiceName = callingServiceName;
+  }
+
+  public AuthFilter() {
+    this.callingServiceName = "";
+  }
+
+  public Response filter(FilterableRequestSpecification requestSpec,
+      FilterableResponseSpecification responseSpec, FilterContext ctx) {
+
+    if (!requestSpec.getHeaders().hasHeaderWithName("Authorization")) {
+
+      // Token retrieved from some code invoked here
+      String token = "somesupersecrettoken";
+      requestSpec.header(new Header("Authorization", token));
+    }
+
+    final Response response = ctx.next(requestSpec, responseSpec);
+
+    return response;
+  }
+}
+```
+
+There are two important things to look at here: (1) the constructor which takes an argument @ roughly line 5, and (2) the filter method @ roughly line 14.  
+
+Let's look at the filter method first.  This is a Rest Assured built in method that is available to all classes that implement the Filter class.  This method will contain all of the code you require, including references to external classes and libraries, in order to add information to your call that will make it able to be authorized.  In this sample code, the filter checks to see if there is already an "Authorization" header in the RestAssured RequestSpec, and if there isn't, it adds one with "supersecrettoken" as the value.  In the real world, there would be more  code here in order to authenticate and retrieve a token to put in as the value.
+
+The Constructor which takes an argument, allows you to pass in service specific data at runtime in order to be able to re-use this filter for multiple services, if you require that.
+
+## Applying the Sample Filter
+
+In order to apply this filter we simply need to instantiate it, and pass it to our Ride call as shown in the [Passing a Filter to a Ride Call](#passing-a-filter-to-a-ride-call) section above, but let's take a look at the sample code:
+
+```
+public class SampleServiceController extends RestApiController {
+  
+  **public static final Filter filter = new AuthFilter("SampleService");**
+  
+  ...
+  
+  private static Response callCore(ObjectType type, String objectPath, ModelObject object,
+      ResponseSpecification expectedResponse, Method method, boolean addAuthorization) {
+    
+    ...
+    
+    Response response = null;
+    if (addAuthorization) {
+      response = fireRestCall(Service.SAMPLE_SERVICE.toString(), objectPath, reqBuilder,
+          expectedResponse, method, **filter**);
+    } else {
+      response = fireRestCall(Service.SAMPLE_SERVICE.toString(), objectPath, reqBuilder,
+          expectedResponse, method);
+    }
+    return response;
+  } 
```

Comment:
maybe another ellipsis and definitely a closing bracket

Reply:
Fixed.